pkg/apis/pipeline/v1/matrix_types.go

```diff
@@ -147,3 +148,39 @@ func (m *Matrix) validateParamTypes() (errs *apis.FieldError) {
 	}
 	return errs
 }
+
+// validatePipelineParametersVariablesInMatrixParameters validates all pipeline paramater variables including Matrix.Params and Matrix.Include.Params
+// that may contain the reference(s) to other params to make sure those references are used appropriately.
+func (m *Matrix) validatePipelineParametersVariablesInMatrixParameters(prefix string, paramNames sets.String, arrayParamNames sets.String, objectParamNameKeys map[string][]string) (errs *apis.FieldError) {
+	if m.hasInclude() {
+		for _, include := range m.Include {
+			for idx, param := range include.Params {
+				stringElement := param.Value.StringVal
+				errs = errs.Also(validateStringVariable(stringElement, prefix, paramNames, arrayParamNames, objectParamNameKeys).ViaFieldIndex("", idx).ViaField("matrix.include.params", ""))
+			}
+		}
+	}
+	if m.hasParams() {
+		for _, param := range m.Params {
+			for idx, arrayElement := range param.Value.ArrayVal {
+				errs = errs.Also(validateArrayVariable(arrayElement, prefix, paramNames, arrayParamNames, objectParamNameKeys).ViaFieldIndex("value", idx).ViaFieldKey("matrix.params", param.Name))
+			}
+		}
```

Comment:
I'm a little bit confused by this validation, I guess it mainly due to I lack the knowledge of matrix and include. 

Is it possible to use array params in include param and string param for matrix param? Why do we only validate string param for include and array params for matrix?

Reply:
Maybe just add a comment here to explain

Reply:
Yes, that is true! Sorry for the confusion, added a comment to specify type required.